### PR TITLE
[6.5.x] DROOLS-2263 : Unexpected results in GDST when using enumerations with commas

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-2263

### CODE REVIEW

**Comment by Rikkola** [comment]
@manstis and @jomarko If you two could please review.

**Comment by manstis** [comment]
@Rikkola to make life easy... is this a cherry-pick?

**Comment by Rikkola** [comment]
@manstis Nope. The fix for master and others is coming, but the code bases are too different.

**Comment by Rikkola** [comment]
Too different to cherry-pick to any direction. This fixes a minor issue where clicking the "a,b" value in the cell would clear the cell value.

**Comment by jomarko** [comment]
Manual check pending, some coverage increasement here https://github.com/Rikkola/kie-wb-common/pull/2

**Comment by jomarko** [comment]
The manual check don't pass :(
If I reopen table [1] and double click the cell with complex enumerations are still not selected [2]

[1]
screenshot from 2018-04-25 15-59-23
[2]
screenshot from 2018-04-25 15-59-30

**Comment by Rikkola** [comment]
@jomarko Well spotted. Based on the build queues taking over a day I doubt that alone would have blocked this. I'll take a look at this tomorrow and it will be ready for the next go.

**Comment by Rikkola** [comment]
Ok so the enumeration file acts differently than the in dtable enumerations I tested and fixed.

**Comment by manstis** [comment]
@Rikkola Is this ready for re-review?

**Comment by Rikkola** [comment]
@manstis Not yet. Still having issues with enums from files.

**Comment by mareknovotny** [comment]
I would add here it is for https://issues.jboss.org/browse/RHBRMS-2754

**Comment by Rikkola** [comment]
@manstis @jomarko @mbiarnes Ok. Ready for second round of reviews with the fix for the issue above and with extra test coverage from @jomarko .

**Comment by mareknovotny** [comment]
@jomarko great, we will wait to all tests pass and merge it

File: kie-wb-common-widgets/kie-wb-decorated-grid-widget/src/main/java/org/kie/workbench/common/widgets/decoratedgrid/client/widget/cells/AbstractPopupEditCell.java [hunk 1]
```diff
@@ -177,4 +178,19 @@ protected abstract void startEditing( final Context context,
                                           final Element parent,
                                           final C value );
 
+    protected String[] splitPreserveQuotes(final String quoteCharacter,
```
Comment (manstis) [inline kie-wb-common-widgets/kie-wb-decorated-grid-widget/src/main/java/org/kie/workbench/common/widgets/decoratedgrid/client/widget/cells/AbstractPopupEditCell.java]: Any takers for correcting formatting?

File: kie-wb-common-widgets/kie-wb-decorated-grid-widget/src/test/java/org/kie/workbench/common/widgets/decoratedgrid/client/widget/cells/PopupDropDownEditCellMultipleSelectTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
```
Comment (manstis) [inline kie-wb-common-widgets/kie-wb-decorated-grid-widget/src/test/java/org/kie/workbench/common/widgets/decoratedgrid/client/widget/cells/PopupDropDownEditCellMultipleSelectTest.java]: Want to update?

**Comment by Rikkola** [comment]
Thanks for the review. Fixed the issues @manstis found.

**Comment by Rikkola** [comment]
These failures make no sense.

Jenkins please retest this.

**Comment by mbiarnes** [comment]
Jenkins please retest this.

**Comment by Rikkola** [comment]
Same two tests failing. Both related to Whitelist save, but that should have nothing to do with this.

Jenkins please retest this.

**Comment by Rikkola** [comment]
Same test failing again. Works locally in CLI and in Idea.

Jenkins please retest this.

**Comment by Rikkola** [comment]
Or don't test. Looks like browser had not refreshed even after I rebooted the computer. Refreshed when I commented. "All checks have passed"

This is ready to go in.

**Pull request merged**